src/care/ProfilePlugin.cpp

```diff
@@ -0,0 +1,59 @@
+#include "care/ProfilePlugin.h"
+
+#include "care/PluginData.h"
+
+/* CUDA profiling macros */
+#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT
+#include "nvToolsExt.h"
+#endif
+
+namespace care{
+   unsigned int ProfilePlugin::s_current_color = 0;
+   uint32_t ProfilePlugin::s_colors[7] = { 0x0000ff00, 0x000000ff, 0x00ffff00, 0x00ff00ff, 0x0000ffff, 0x00ff0000, 0x00ffffff };
+   int ProfilePlugin::s_num_colors = sizeof(s_colors) / sizeof(uint32_t);
+   bool ProfilePlugin::s_profile_host_loops = true; 
+   
+   ProfilePlugin::ProfilePlugin() {}
+   
+   void ProfilePlugin::registerPlugin() {
+      static RAJA::util::PluginRegistry::add<care::ProfilePlugin> L ("Profile plugin", "Care plugin for profiling");
+   }
+
+   void ProfilePlugin::preLaunch(const RAJA::util::PluginContext& p) {
+#if CARE_HAVE_NVTOOLSEXT
+#if defined(__CUDACC__)
+      // Profile the host loops
+      if (s_profile_host_loops) {
+         if (p.platform == RAJA::Platform::host) {
+            std::string name = PluginData::getFileName() + std::to_string(PluginData::getLineNumber());
+            int color_id = s_current_color + 1;
+            color_id = color_id % s_num_colors;
+         
+            // TODO: Add error checking
+            nvtxEventAttributes_t eventAttrib = { 0 };
+            eventAttrib.version = NVTX_VERSION;
+            eventAttrib.size = NVTX_EVENT_ATTRIB_STRUCT_SIZE;
+            eventAttrib.colorType = NVTX_COLOR_ARGB;
+            eventAttrib.color = s_colors[color_id];
+            eventAttrib.messageType = NVTX_MESSAGE_TYPE_ASCII;
+            eventAttrib.message.ascii = name.c_str();
+            nvtxRangePushEx(&eventAttrib);
+         }
+      }
+#endif // CARE_HAVE_NVTOOLSEXT
+#endif // defined(CARE_GPUCC)
+   }
+
+
+   void ProfilePlugin::postLaunch(const RAJA::util::PluginContext& p) {
+#if defined(__CUDACC__)
```

Comment:
`#if defined(__CUDACC__) && CARE_HAVE_NVTOOLSEXT`

Reply:
Did you push up your changes? This doesn't look resolved to me. Also, I prefer that you just make a comment to say if something is fixed and let a reviewer click the "Resolve conversation" button after they have checked that it is resolved.

Reply:
got it! also i made all the changes but i think there were problems involving swp files and some of my changes got reverted. i'll look through your comments again to make sure i didn't miss anything else.

Reply:
I think you put the fix in the wrong spot.